src/crawler.ts

```diff
@@ -796,7 +796,20 @@ self.__bx_behaviors.selectMainBehavior();
     const { page, cdp, data, workerid, callbacks, directFetchCapture } = opts;
     data.callbacks = callbacks;
 
-    const { url } = data;
+    const { url, seedId } = data;
+
+    const auth = this.seeds[seedId].auth;
+
+    if (auth) {
+      logger.debug("Setting HTTP basic auth for seed", {
+        seedId,
+        seedUrl: this.seeds[seedId].url,
+      });
+      await page.setExtraHTTPHeaders({ Authorization: "Basic " + auth });
+      opts.isAuthSet = true;
+    } else if (opts.isAuthSet) {
+      await page.setExtraHTTPHeaders({});
+    }
```

Comment:
Can you talk me through what's happening here? I don't see `options.isAuthSet` being used in the worker (unless for logging?) after being set, and unclear why the need for the else if/adding empty headers.

Reply:
oh, this is just an optimization to avoid calling `page.setExtraHTTPHeaders({})` if they were never set before. Sort of unnecessary, can juts always call it - its only being set / checked here.